Item rando changes

erudnick-cohen opened the pull request:
Changes to provide support for the full and key item randomizers

anmart commented:
There's a big error in wram.asm. I made some assumptions about unions that I shouldn't have and shifted a lot of wram labels into incorrect locations. Closing until I fix the ENDUs

anmart commented:
Turns out that the last union in all these cases is always large enough because it's just 1:1 to line up with the code after the union (the only plus of `-` syntax)

erudnick-cohen committed:
update gitignore

erudnick-cohen committed:
Remove the rocket who can block the flower shop during the rocket takeover

erudnick-cohen committed:
Flower shop gives item even after sudowoodo defeated

erudnick-cohen committed:
make red scale verbosegiveitem and removed unneeded text box

erudnick-cohen committed:
Switched mystery egg to verbosegiveitem and removed associated text box

erudnick-cohen committed:
Force SS Aqua to operate every day of the week

erudnick-cohen committed:
Revert "Force SS Aqua to operate every day of the week"

This reverts commit 7ec7151f9980b5bdaaf0a4dc32fe12c182401990.

erudnick-cohen committed:
Revert "Switched mystery egg to verbosegiveitem and removed associated text box"

This reverts commit ad25e808dc09973fd74489bf755d07b358f37241.

erudnick-cohen committed:
Revert "update gitignore"

This reverts commit 44546b6118472d24fd1b214423e6d4e456e7496f.

erudnick-cohen committed:
Revert "Remove the rocket who can block the flower shop during the rocket takeover"

This reverts commit b80a0a177b05b4b1b44f04b5b6f0beef3a2de13c.

erudnick-cohen committed:
Revert "make red scale verbosegiveitem and removed unneeded text box"

This reverts commit 2677390f674184c49c8b73a95e3cb5539a531e43.

erudnick-cohen committed:
Revert "Flower shop gives item even after sudowoodo defeated"

This reverts commit d44f8f22b9275beca073cd401b4a0003bbb7b0f6.

erudnick-cohen committed:
Make officer at lab optional to prevent scene change making pokeballs unobtainable

Officer can still be spoken too if you want to name the rival (or can just be skipped if you don't)

erudnick-cohen committed:
Ensure player can still obtain secretpotion after giving it to jasmine

erudnick-cohen committed:
Switched lost item to verbosegiveitem and removed assoicated text

erudnick-cohen committed:
Make director permanently stuck in basement and have him check if team rocket is attacking before he gives out an item

Director will remain in basement after team rocket is defeated and his item can still be obtained at that point. However, his item cannot be obtained prior to the rocket invasion occuring.

erudnick-cohen committed:
Radio card still available during rocket takeover

erudnick-cohen committed:
Buena's item still available during rocket takeover

erudnick-cohen committed:
Fight fake director or rocket boss in any order

NOTE: there are now two places in the code where basement key and clear bell are given out with this!

erudnick-cohen committed:
Only need to talk to power plant manager once and machine part and zap cannon always available

Note: talking to the power plant manager a second time will respawn the rocket in cerulean gym, which will also respawn the machine part.

erudnick-cohen committed:
Reroute blackthorn and goldenrod already beaten checks to mt silver flypoint

This ensures that you don't get locked out of these badges by obtaining the vanilla badge for these locations. Mt. silver flypoint is used because it requires access to all the badges in the first place.

erudnick-cohen committed:
Switched kanto gym leaders to check if they are beaten, instead of checking if you have their vanilla badge

erudnick-cohen committed:
Make dragon fang a more normal item so it doesn't cause harmless trippy graphics errors when it would be switched to verbosegiveitem

erudnick-cohen committed:
Train can be taken without restoring power to kanto

Note that pass cannot normally be obtained unless you have already restored power... unless you're playing item rando.

erudnick-cohen committed:
Before checking for everstone, elm needs to actually check if you have the egg

Otherwise it can get skipped, which then skips the pokeballs given by the aide

erudnick-cohen committed:
Fixed power plant to require talking director once with no possibility of infinitely spawning rockets or machine parts

erudnick-cohen committed:
Made director check actually work correctly

erudnick-cohen committed:
Made dragon fang actually just a perfectly normal itemball

erudnick-cohen committed:
Lost item doesn't require talking to copycat, so that you don't softlock by turning it in before checking it

erudnick-cohen committed:
Merge pull request #1 from erudnick-cohen/ItemRandoSupportTesting

Merging in fixes from actually testing the rom with changes and a few improvements

erudnick-cohen committed:
These changes should ONLY ever be used for testing

erudnick-cohen committed:
Update .gitignore

erudnick-cohen committed:
Fixed up blackthorn to use verbose give item and to have clair just check on giving you the tm to see if you have the badge already

Technically you can't fail to obtain the TM...except in full item randomizer (where this can happen since it might not be a TM). However, there is no harm in clair giving you a badge for a "second time" so this shouldn't really be an issue.

erudnick-cohen committed:
Removed whitney's badge check so that it just checks if you recieved attract

This can technically fail in full item rando, but its otherwise harmless (getting a badge twice doesn't actually do anything).

erudnick-cohen committed:
Merge branch 'master' into ItemRandoSupportTesting

erudnick-cohen committed:
Revert "Merge branch 'master' into ItemRandoSupportTesting"

This reverts commit 5f35df1b0982dba2161309741957b74fe46cf3e3, reversing
changes made to 0a67dab01d35f9a58feffc31e98924bc3ec6490c.

erudnick-cohen committed:
Changes from master for testing

erudnick-cohen committed:
Try to make Ho-Oh chamber accesible by beating Ho-Oh. Also, make the rainbow wing obtainable by beating suicune

erudnick-cohen committed:
Merge pull request #2 from erudnick-cohen/DrasticChanges

Try to make Ho-Oh chamber accesible by beating Ho-Oh. Also, make the …

erudnick-cohen committed:
Ho-Oh shouldn't appear if you haven't beaten the E4 yet

erudnick-cohen committed:
Made ho oh chamber actually work and not randomly sometimes lock the door behind you

talking to the wall will always open the door

erudnick-cohen committed:
Getting the vanilla rainbow wing requires defeating E4

erudnick-cohen committed:
Comments in ho-oh chamber

erudnick-cohen committed:
Tin tower comments

erudnick-cohen committed:
Revert "These changes should ONLY ever be used for testing"

This reverts commit 0a67dab01d35f9a58feffc31e98924bc3ec6490c.

erudnick-cohen committed:
Reinclude flower shop skip in changes

erudnick-cohen committed:
Merge pull request #3 from erudnick-cohen/ChangesWithComments

TODO comments for where flags definitely should be added and tin tower and ho-oh chamber related changes
